src/types/resolveStatements.ts

```diff
@@ -353,20 +353,42 @@ function processStatements(
                     // Check type
                     const expressionType = getExpType(ctx, s.expression);
                     const tailType = getExpType(ctx, s.path);
-                    // Check if types are Int
+
+                    // Check if any of the types is not ref or is optional or types themselves don't match
                     if (
                         expressionType.kind !== "ref" ||
-                        expressionType.name !== "Int" ||
                         expressionType.optional ||
                         tailType.kind !== "ref" ||
-                        tailType.name !== "Int" ||
                         tailType.optional
                     ) {
                         throwCompilationError(
-                            `Type error: Augmented assignment is only allowed for Int type`,
+                            `Type error: Augmented assignment ${s.op}= is only allowed for Int and Bool types`,
                             s.loc,
                         );
                     }
+
+                    if (expressionType.name !== tailType.name) {
+                        throwCompilationError(
+                            `Type error: Augmented assignment ${s.op}= is only allowed for matching types`,
+                            s.loc,
+                        );
+                    }
```

Comment:
Basically, the operator should take priority in error messages. So, we first check if the l-value type and the operator do not match. And only if those match but the right-hand side expression does not, we complain about the expression type.